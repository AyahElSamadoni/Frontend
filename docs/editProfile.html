<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: editProfile</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: editProfile</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>editProfile<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">This is a description of the editProfile Component.</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="editProfile"><span class="type-signature"></span>new editProfile<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Containers_Profile_EditProfile.js.html">Containers/Profile/EditProfile.js</a>, <a href="Containers_Profile_EditProfile.js.html#line18">line 18</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="state"><span class="type-signature"></span>state<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>editProfileForm</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">The Signup Form object that contains all the form's components properties.
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>screen_name</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">User's screen name properties and configurations.
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>elementType</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The type of the input element.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>elementConfig</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">The input element configurations.
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>type</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The input element type.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>placeholder</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The input element's placeholder.</td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    

        <tr>
            
                <td class="name"><code>value</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The value of the Screen Name.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>valid</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">indicates whether the input value is valid or not.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>touched</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">indicates whether the the input element has been touched or not.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>autoFocus</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">Is set to true when the element should be autofocused.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>errorMessage</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The error message to be displayed when the input is invalid.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>validation</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">The validations required for the screen name.
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>required</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">Indicates whether the field is necessary or not to signup.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>nospace</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">Is set to true when spaces are not allowed in the input value.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>startLetter</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">Is set to true when the input value should start with a letter only.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>maxLength</code></td>
            

            <td class="type">
            
                
<span class="param-type">integer</span>


            
            </td>

            

            

            <td class="description last">The maximum length for the input value.</td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    

        <tr>
            
                <td class="name"><code>name</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">User's name properties and configurations.
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>elementType</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The type of the element.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>elementConfig</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">The input element configurations.
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>type</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The input element type.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>placeholder</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The input element's placeholder.</td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    

        <tr>
            
                <td class="name"><code>value</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The value of the Username.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>valid</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">indicates whether the input value is valid or not.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>touched</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">indicates whether the the input element has been touched or not.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>errorMessage</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The error message to be displayed when the input is invalid.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>validation</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">The validations required for the name.
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>required</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">Indicates whether the field is necessary or not to signup.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>maxLength</code></td>
            

            <td class="type">
            
                
<span class="param-type">integer</span>


            
            </td>

            

            

            <td class="description last">The maximum length for the input value.</td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    

        <tr>
            
                <td class="name"><code>bio</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">User's bio properties and configurations.
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>elementType</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The type of the element.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>elementConfig</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">The input element configurations.
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>type</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The input element type.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>placeholder</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The input element's placeholder.</td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    

        <tr>
            
                <td class="name"><code>value</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The value of the Username.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>valid</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">indicates whether the input value is valid or not.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>touched</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">indicates whether the the input element has been touched or not.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>errorMessage</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The error message to be displayed when the input is invalid.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>validation</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">The validations required for the bio.
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>required</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">Indicates whether the field is necessary or not to signup.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>maxLength</code></td>
            

            <td class="type">
            
                
<span class="param-type">integer</span>


            
            </td>

            

            

            <td class="description last">The maximum length for the input value.</td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    

        <tr>
            
                <td class="name"><code>loc</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">location of user.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>formIsValid</code></td>
            

            <td class="type">
            
                
<span class="param-type">bool</span>


            
            </td>

            

            

            <td class="description last">indicates whether the form is valid or not before submitting.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>loading</code></td>
            

            <td class="type">
            
                
<span class="param-type">bool</span>


            
            </td>

            

            

            <td class="description last">indicates whether the spinner should be displayed or not.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>file</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">image file path.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>imapePreview</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">default image url.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>currentProfile</code></td>
            

            <td class="type">
            
                
<span class="param-type">JSON</span>


            
            </td>

            

            

            <td class="description last">data of current user.</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Containers_Profile_EditProfile.js.html">Containers/Profile/EditProfile.js</a>, <a href="Containers_Profile_EditProfile.js.html#line68">line 68</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="checkValidity"><span class="type-signature"></span>checkValidity<span class="signature">(value, rules)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Check if a value matches a given set of validation rules.
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>value</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">The value to be checked.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>rules</code></td>
            

            <td class="type">
            
                
<span class="param-type">json</span>


            
            </td>

            

            

            <td class="description last">The set of vaidation rules to check against.</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Containers_Profile_EditProfile.js.html">Containers/Profile/EditProfile.js</a>, <a href="Containers_Profile_EditProfile.js.html#line169">line 169</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="inputChangeHandler"><span class="type-signature"></span>inputChangeHandler<span class="signature">(event, inputIdentifier)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Handles changes in the input elements by running its validity checks and updating its value in the state.
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>event</code></td>
            

            <td class="type">
            
                
<span class="param-type">event</span>


            
            </td>

            

            

            <td class="description last">The change event.</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>inputIdentifier</code></td>
            

            <td class="type">
            
                
<span class="param-type">integer</span>


            
            </td>

            

            

            <td class="description last">The input element's identifier.</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Containers_Profile_EditProfile.js.html">Containers/Profile/EditProfile.js</a>, <a href="Containers_Profile_EditProfile.js.html#line134">line 134</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="editProfile.html">editProfile</a></li><li><a href="forgotPassword.html">forgotPassword</a></li><li><a href="login.html">login</a></li><li><a href="Newsfeed.html">Newsfeed</a></li><li><a href="profile.html">profile</a></li><li><a href="signup.html">signup</a></li></ul><h3>Global</h3><ul><li><a href="global.html#button">button</a></li><li><a href="global.html#input">input</a></li><li><a href="global.html#LandingPage">LandingPage</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Tue May 07 2019 23:43:42 GMT+0200 (Egypt Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>