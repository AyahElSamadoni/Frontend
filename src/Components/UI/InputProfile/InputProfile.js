import React from "react";
import "./InputProfile.css";

const inputProfile = props => {
  let inputElementt = null;
  let inputClasses = "form-control InputElement";
  let validationError = null;
  if (props.invalid && props.touched) {
    inputClasses = "InputElement Invalid";
    validationError = <p className="ValidationError"> {props.errorMessage}</p>;
  } else if (props.invalidScreenname) {
    inputClasses = "InputElement Invalid";
    validationError = (
      <p className="ValidationError">
        {" "}
        "invalid Screenname. Screenname already exist"
      </p>
    );
  } else if (props.errorScreenname) {
  } else if (props.invalidLenScreenname) {
    inputClasses = "InputElement Invalid";
    validationError = (
      <p className="ValidationError">
        {" "}
        "Screenname must be not less than 3 characters and not more than 15
        characters"
      </p>
    );
  } else validationError = null;
  inputElementt = (
    <input
      className={inputClasses}
      {...props.elementConfig}
      value={props.value}
      onChange={props.changed}
      autoFocus={props.autoFocus}
    />
  );

  //<label className={classes.Label}>{props.label}</label>
  return (
    <div className="form-group">
      {inputElementt}
      {validationError}
    </div>
  );
};
export default inputProfile;
