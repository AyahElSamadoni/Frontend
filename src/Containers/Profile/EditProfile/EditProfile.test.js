import React from "react";
import { shallow, configure } from "enzyme";
import editProfile from "./EditProfile";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import InputProfile from "../../../Components/UI/InputProfile/InputProfile";

configure({ adapter: new Adapter() });
describe("<editProfile>", () => {
  it("should be defined", () => {
    expect(<editProfile />).toBeDefined();
  });
  it("should render", () => {
    const wrapper = shallow(<editProfile />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should render", () => {
    const wrapper = shallow(<editProfile />);
    expect(wrapper).toMatchSnapshot();
  });

  test("pass a selected value to handleChange handler to handle location", () => {
    const value = "Afghanistan";
    const _handleChange = jest.fn();
    const wrapper = shallow(
      <select onChange={_handleChange} className="form-control" />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.find("select").simulate("change", "Afghanistan");

    expect(_handleChange).toBeCalledWith(value);
  });

  //   it("Should update the email value state when an email is given", () => {
  //     const wrapper = shallow(<editProfile />);
  //     const name = "mirna";
  //     const check = wrapper
  //       .find(InputProfile)
  //       .at(0)
  //       .dive()
  //       .find("input");
  //     check.simulate("change", {
  //       target: {
  //         value: name
  //       }
  //     });

  //     expect(wrapper.state().editProfileForm.screen_name.value).toBe(name);
  //   });
});
