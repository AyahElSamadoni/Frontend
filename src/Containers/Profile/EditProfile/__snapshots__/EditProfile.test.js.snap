// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<editProfile> pass a selected value to handleChange handler to handle location 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <select
    className="form-control"
    onChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "form-control",
      "onChange": [MockFunction],
    },
    "ref": null,
    "rendered": null,
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "form-control",
        "onChange": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<editProfile> render a small label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <label
    className="custom-file-label"
    for="inputGroupFile01"
  >
    Choose file
  </label>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Choose file",
      "className": "custom-file-label",
      "for": "inputGroupFile01",
    },
    "ref": null,
    "rendered": "Choose file",
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Choose file",
        "className": "custom-file-label",
        "for": "inputGroupFile01",
      },
      "ref": null,
      "rendered": "Choose file",
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<editProfile> should render 1`] = `<editProfile />`;

exports[`<editProfile> should render 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <editProfile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "editProfile",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "editProfile",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<editProfile> should render 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "btn btn-primary signupButton",
      "disabled": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "btn btn-primary signupButton",
        "disabled": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
